### PR TITLE
Phase1 mapper changes

### PR DESCRIPTION
**Pull request by resmohan**

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -8,7 +8,7 @@
       </list>
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_21" project-jdk-name="11" project-jdk-type="JavaSDK">
```
Comment (prinjal) [inline .idea/misc.xml:11]: Looks good! Thanks for changing the Java version.

File: src/main/java/Main.java [hunk 2]
```diff
@@ -1,6 +1,36 @@
+import java.io.IOException;
```
Comment (prinjal) [inline src/main/java/Main.java:1]: Looks good!

File: src/main/java/model/PatternInfo.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package model;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import org.apache.hadoop.io.WritableComparable;
+
+public class PatternInfo implements WritableComparable<PatternInfo> {
```
Comment (ghost) [inline src/main/java/model/PatternInfo.java:8]: Does this mean that you have already implemented the Group comparator? I believe there is no need to write a separate group comparator now, right?

File: src/main/java/model/SentimentInfo.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  @Override
+  public void write(DataOutput dataOutput) throws IOException {
+    dataOutput.writeUTF(sentimentType.name());
+  }
+
+  @Override
+  public void readFields(DataInput dataInput) throws IOException {
+    sentimentType = SentimentType.valueOf(dataInput.readUTF());
+  }
+
```
Comment (ghost) [inline src/main/java/model/SentimentInfo.java:27]: We might need a getter for retrieving sentiment type object

File: src/main/java/model/PatternInfo.java (hunk 3 above)
Comment (resmohan) [reply src/main/java/model/PatternInfo.java:8]: Probably yes. may be we can just write the reducer and test once. If it is working as expected, then all good.

File: src/main/java/model/SentimentInfo.java (hunk 4 above)
Comment (resmohan) [reply src/main/java/model/SentimentInfo.java:27]: If you need in reducer/in any of your codes, feel free to add it then.

File: src/main/java/model/PatternInfo.java (hunk 3 above)
Comment (ghost) [reply src/main/java/model/PatternInfo.java:8]: Sounds good!!

File: src/main/java/model/SentimentInfo.java (hunk 4 above)
Comment (prinjal) [reply src/main/java/model/SentimentInfo.java:27]: Can we please add it to this branch? so that we keep modifying things seperately.

**Review by ghost — APPROVED** [review]
Looks good!

**Pull request merged**